rectools/visuals/metrics_app.py

```diff
@@ -0,0 +1,128 @@
+        auto_display: bool = True,
+        color_discrete_sequence: tp.Optional[tp.List[str]] = None,
+    ):
+        self.metrics_data = df_metrics_data
+        self.chart_width = chart_width
+        self.chart_height = chart_height
+        self.show_legend = show_legend
+        self.auto_display = auto_display
+        self.color_discrete_sequence = (
+            px.colors.qualitative.Plotly if color_discrete_sequence is None else color_discrete_sequence
+        )
+
+        self._validate_input_data()
+        self._validate_input_patameters()
+        if self.auto_display:
+            self.display()
+
+    @classmethod
+    def construct(
+        cls,
+        df_metrics_data: pd.DataFrame,
+        chart_width: tp.Union[float, int] = 800,
+        chart_height: tp.Union[float, int] = 600,
+        show_legend: bool = False,
+        auto_display: bool = True,
+        color_discrete_sequence: tp.Optional[tp.List[str]] = None,
+    ) -> "MetricsApp":
+        """TODO: add docstring"""
+        return cls(df_metrics_data, chart_width, chart_height, show_legend, auto_display, color_discrete_sequence)
+
+    def _validate_input_data(self) -> None:
+        if not isinstance(self.metrics_data, pd.DataFrame):
+            raise ValueError("Incorrect input type. `metrics_data` should be a DataFrame")
+        if Columns.Split not in self.metrics_data.columns:
+            raise KeyError("Missing 'Split' column in `metrics_data` DataFrame")
+        if Columns.Model not in self.metrics_data.columns:
+            raise KeyError("Missing 'Model' column in `metrics_data` DataFrame")
+        if len(self.metrics_data.columns) < 3:
+            raise KeyError("`metrics_data` DataFrame assumed to have at least one metric column")
+
+    def _validate_input_patameters(self) -> None:
+        if self.chart_width < 10:
+            raise ValueError(
+                "Incorrect `chart_width` value. `chart_width` should be a float or int in the interval [10, inf]"
+            )
+        if self.chart_height < 10:
+            raise ValueError(
+                "Incorrect `chart_height` value. `chart_height` should be a float or int in the interval [10, inf]"
+            )
+
+    def _get_metric_names(self) -> tp.List[str]:
+        non_metric_columns = {Columns.Split, Columns.Model}
+        return [col for col in self.metrics_data.columns if col not in non_metric_columns]
+
+    def _get_folds_number(self) -> int:
+        return self.metrics_data[Columns.Split].nunique()
+
+    def _make_chart_data(self, fold_number: int) -> pd.DataFrame:
+        return self.metrics_data[self.metrics_data[Columns.Split] == fold_number].drop(columns=Columns.Split)
+
+    def _make_chart_data_avg(self) -> pd.DataFrame:
+        return self.metrics_data.drop(columns=Columns.Split).groupby(Columns.Model, sort=False).mean().reset_index()
+
+    def _create_chart(self, avg_folds: bool, fold_number: int, metric_x: str, metric_y: str) -> go.Figure:
```

Comment:
1. You don't need `fold_number` if `avg_folds is True`, so let it be optional
2. It's better to order arguments by importance, so here it should be `metric_x, metric_y, avg_folds, fold_number`

Reply:
Done, but may require additional discussion due to refactoring